elastic/kibana
# [Uptime] Update snapshot counts

## Summary

Resolves https://github.com/elastic/kibana/issues/47376.

~Code is WIP.~

The goal of this patch is to utilize the monitor iterator to ensure that the snapshot counts we're showing the user correspond to the values in the monitor list.

In addition to updating how we compute these values, we're moving the querying code from the existing GQL schema to a REST endpoint with `io-ts`-generated typing for validation. Additionally, we've begun a state management migration from internal component state and context reliance to a dedicated Redux-powered solution. As a result, there're some added functions around data fetching, state reduction, and action dispatching. It also resulted in the `Snapshot` component becoming a `connect`ed container.

## Testing this PR
0. Start up one or more Heartbeat locations and run this patch
1. Ensure that the counts in the Snapshot view at the top of the _Overview_ page correspond to the count of each status type of monitor shown in the table
2. When you filter your environment, ensure that the filters you're applying result in the same numeric changes between the monitor list and the snapshot count

The goal here is to ensure that the counts we show our users correspond to the actual data we're displaying in our table.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
